Added installation file

Elliptic461 opened the pull request:
Add an installation notebook. Include environemnt setup with conda.

Elliptic461 committed:
Install instruction + installation

Add instruction for how to get vizfold working.

Elliptic461 committed:
Delete install.md

Elliptic461 committed:
Delete install.ipynb

Elliptic461 committed:
Add files via upload

Elliptic461 committed:
Delete install.ipynb

Elliptic461 committed:
Add installation

Elliptic461 committed:
Delete install.ipynb

Elliptic461 committed:
Add files via upload

Elliptic461 commented:
Updated install.ipynb to account for people who aren't using PACE ICE Cluster. Also added a small readme file that tells you how to run install.ipynb and the requirements. The readme file and install.ipynb is put in a installationGuide folder to help user easily find the install.ipynb.

Vishram1123 committed:
Use Bash cell magic instead of Bash kernel, and add Conda installation directory

The cell magic change is so the need for the Bask kernel is removed. Instead, a generic Python kernel can be used instead. The Conda install dir change is so different directories can be used if your home directory is too small

Vishram1123 committed:
Revise README with updated installation steps and paths

Updated installation instructions for AlphaFold, including new data directory paths and changes to the installation process.

Elliptic461 committed:
Merge pull request #1 from Test-VR1123/main

Update install.ipynb and README.md

katamyra commented:
I tested this out and I noticed that the git clone commands will create a new folder called "~/scratch" inside scratch, instead of recognizing the "~" as the root directory, has anyone else noticed this?

Also thoughts on using ```os.environ['ROOT_DIR'] = os.path.expanduser('~/scratch')```?

katamyra commented:
Also curious, how long does `conda env create -n openfold_env -f environment.yml` take you guys to run on PACE? I am testing replacing all these commands with `mamba` which should be faster.

Vishram1123 commented:
> I tested this out and I noticed that the git clone commands will create a new folder called "~/scratch" inside scratch, instead of recognizing the "~" as the root directory, has anyone else noticed this?
> 
> Also thoughts on using `os.environ['ROOT_DIR'] = os.path.expanduser('~/scratch')`?

This would probably be smart, but it might be better to keep the os.environ having just strings, and use the os.path.expanduser later on to fix the paths

Elliptic461 committed:
Fix paths.

Vishram1123 committed:
Add files via upload

Vishram1123 committed:
Add pip install command for dependencies

Vishram1123 committed:
Add comment for pip dependencies installation

Vishram1123 committed:
Fix formatting in installation guide notebook

Elliptic461 committed:
Merge pull request #2 from Test-VR1123/main

Add Mamba solver

katamyra commented:
@smarru would it be possible to test this and see if its mergeable for others in the class to use for easy setup?
- Krish Katariya

Elliptic461 commented:
> @Elliptic461 I just tried testing everything from a new clean environment, and everything works great! Thanks for fixing the issue I pointed out and getting mamba set up for speedup. Do you think maybe we could also add downloading the alphafold parameters to the end of the install script?

Glad to hear that it works. As for the adding download alphafold parameters, let me try that.

Dfromond commented on installationGuide/README.md:
```diff
@@ -0,0 +1,22 @@
+
+- Conda package manager 
+- Access to the AlphaFold data directory (typically at `/storage/ice1/shared/d-pace_community/alphafold/alphafold_2.3.2_data` for Georgia Tech PACE ICE, and `/anvil/datasets/alphafold/db` for Purdue Anvil)
+- Sufficient disk space
```
Please elaborate on this section. Estimated disk space necessary? Does Purdue also necessitate Conda be put in a scratch directory instead of home?

Vishram1123 replied:
For the Anvil Conda in Scratch, this is required because when I was trying to run this setup, the installation filled up my home directory, and I didn't have enough space, and I had to use my Scratch directory (of which it used 50 gb) to install.

Dfromond commented on installationGuide/README.md:
```diff
@@ -0,0 +1,22 @@
+## Installation
+
+1. Open the `install.ipynb` notebook in Jupyter
+2. Select a Python kernel
```
I would recommend including a step between this to ensure your conda env is activated and all dependencies are installed. This guide should cater the setup process towards those who are more familiar with protein structures and less familiar with HPCs.

Dfromond reviewed (changes requested):
`README.md` changes. If `install.ipynb` has been validated, that file looks good.

Elliptic461 committed:
Update README.md

Dfromond commented:
Two more points:
1. This PR will need to be `Squash & Merge`d.
2. _This is completely up to you and the group members._ If you would like me to add how HPC environments interact with Visfold, I can update my diagram to include this.

Elliptic461 committed:
Update README.md

Elliptic461 committed:
Update README.md

Elliptic461 committed:
Update README.md

Elliptic461 committed:
Update README.md

Vishram1123 committed:
Update README with clearer Conda and Jupyter instructions

Clarified instructions for loading Conda and using the OpenFold environment in Jupyter.

Elliptic461 committed:
Merge pull request #3 from Test-VR1123/main

Update README with clearer Conda and Jupyter instructions

girip commented on install.md:
These are nice instructions for install. 

It would help to provide better documentation for each step, including a link to where it was derived. I think some parts are from the openfold installation; refer to their installation instructions, with explict step or line number in https://openfold.readthedocs.io/en/latest/Installation.html. 

Also, looks like you are using model from /storage/ice1, which deviates from the openfold installation instructions. Please add documentation about the source of this file, if it was initially got from the same source and then using a local version etc ..

Vishram1123 committed:
Update installation instructions for Jupyter kernel

Elliptic461 committed:
Merge pull request #4 from Vishram1123/patch-1

Update installation instructions for Jupyter kernel

Elliptic461 committed:
Update install.ipynb

Elliptic461 committed:
Update install.ipynb

Elliptic461 committed:
Update install.ipynb

Vishram1123 committed:
Enhance installation script with verbose output

Added verbose flag '-vvv' to various conda and pip commands for detailed output during installation.

Vishram1123 committed:
Fix formatting in installation guide for Conda module

Vishram1123 committed:
Enable verbose logging in installation script

Added verbose logging to Conda activation steps.

Vishram1123 committed:
Reduce verbosity in installation commands

Updated installation commands to reduce verbosity in output.

Elliptic461 committed:
Merge pull request #5 from Vishram1123/patch-2

Enhance installation script with verbose output

Commit:
Load anaconda3 at every cell

Vishram1123 committed:
Add files via upload

Elliptic461 committed:
Merge pull request #6 from Test-VR1123/main

Add files via upload

Elliptic461 committed:
Fix pip args and symlinks

Elliptic461 committed:
Merge branch 'AI2Science:main' into main

Elliptic461 committed:
Pin flash-attn dependency

Elliptic461 committed:
Allow mamba module to be changed

Elliptic461 committed:
Update install.ipynb

Vishram1123 committed:
Move mamba's command into env and add to readme

Vishram1123 committed:
Add converted script

Elliptic461 committed:
Merge pull request #7 from Vishram1123/patch-3

Move mamba's command into env and add to readme

Elliptic461 committed:
Update README.md

Vishram1123 committed:
Replace cli install.py with install.sh

Vishram1123 committed:
Use conda for init instead of mamba

Vishram1123 committed:
Clean up installation script by removing redundant lines

Removed redundant Miniforge module loading and Conda activation steps from the installation script.

Elliptic461 committed:
Merge pull request #8 from Vishram1123/patch-3

Replace cli install.py with install.sh
